packages/axelar-wasm-std/src/msg_id/tx_hash.rs

```diff
@@ -0,0 +1,138 @@
+use core::fmt;
+use std::fmt::Display;
+use std::str::FromStr;
+
+use cosmwasm_std::HexBinary;
+use error_stack::{ensure, Report, ResultExt};
+use lazy_static::lazy_static;
+use regex::Regex;
+
+use super::Error;
+use crate::hash::Hash;
+use crate::nonempty;
+
+pub struct HexTxHash {
+    pub tx_hash: Hash,
+}
+
+impl HexTxHash {
+    pub fn tx_hash_as_hex(&self) -> nonempty::String {
+        HexBinary::from(self.tx_hash)
+            .to_hex()
+            .try_into()
+            .expect("failed to convert tx hash to non-empty string")
+    }
+
+    pub fn new(tx_id: impl Into<[u8; 32]>) -> Self {
+        Self {
+            tx_hash: tx_id.into(),
+        }
+    }
+}
+
+const PATTERN: &str = "^[0-9a-f]{64}$";
```

Comment:
```suggestion
const PATTERN: &str = "^0x[0-9a-f]{64}$";
```
this should start with 0x, sorry I missed this the first time.

The existing tx hash + event index message id starts with 0x, so we should be consistent

Reply:
Done